gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
             coords={"phase": phases, "wlen": spectra.wlen},
         )
 
+    def calc_transit_spectrum(
+            self,
+            mmw,
+            lat_points=4,
+            lon_resolution=10,
+            gravity=None,
+            rplanet=None,
+            pressure_0=None,
+            co_ratio=0.55,
+            feh_ratio=0.0,
+            mass_frac=None,
+    ):
+        """
+        Calculate the transit spectrum. This function avarages T-p profiles in
+        the terminator region and uses poorman equilbriums chemistry.
+
+        Parameters
+        ----------
+        mmw: float or 1D-array
+            Mean molecular weight (in atomic units). Will be globally uniform if
+            float or horizonatally uniform if 1D.
+        lat_points : int
+            Number of latitude points considered per terminator.
+        lon_resolution : float
+            Longitude angle to avarage over at the equator of the terminators. In !degrees!.
+        gravity: float
+            surface gravity in !cgs!. Will default to the value provided by GCMT.
+        rplanet: float
+            planet radius in !cm!. Will default to the value provided by GCMT.
+        pressure_0: float
+            pressure level of the gravity and radius. Will default to the value provided by GCMT.
+        co_ratio: float, optional
+            The C/O ratio. Currently only one global value allowed. Defaults to 0.55.
+        feh_ratio: float, optional
+            The metalicity ratio. Currently only one global value allowed.
+            Defaults to 0.0.
+        mass_frac: List[List[Dict]], optional
+            The mass fractions for each t_p point. Structure is as follows: mass_frac[i][j]['key']
+            - i: [2 elements] 0 for morning terminator, 1 for evening terminator
+            - j: [lat_points elements] latitude point starting from lowest (closeset to lat = -90) to highest (closest to lat = 90)
+            - dict must contain all mass fractions  elements given as opacity species to prt
+
+
+        Returns
+        -------
+        spectra: xr.DataArray
+            Dataarray containing the spectrum.
+            The spectrum is normed to the stellar spectrum.
+
+        """
+        from petitRADTRANS.poor_mans_nonequ_chem import interpol_abundances
+        from petitRADTRANS import nat_cst as nc
+
+        # prepare chemistry input (make them 1D arrays)
+        pres = self.dsi[c["Z"]].values
+        co_ratios = np.ones_like(self.dsi[c["Z"]].values) * co_ratio
+        feh_ratios = np.ones_like(self.dsi[c["Z"]].values) * feh_ratio
+        tlist = [('H2O', 'H2O_HITEMP'),
+                 ('H2O', 'H2O_Exomol_R_1'),
+                 ('CO', 'CO_all_iso_HITEMP'),
+                 ('Na', 'Na_allard'),
+                 ('K', 'K_allard')]
```

Comment:
use an automatic split instead. Its more general. Here is how you can go from linelist to species:
```py
{sp: sp.split('_')[0] for sp in linespecies}
```
You can then do the same thing the other way around to invert the dictionary:
```py
{sp.split('_')[0]: sp for sp in linespecies}
```

Reply:
Thanks! That's exactly what I wanted to do but I did not see.

Reply:
You can get them via prt.line_species or something

Reply:
I found a solution, might not be the prettiest, but it can handle arbitrary input :D